token.go

```diff
@@ -164,36 +166,125 @@ type memoryStore struct {
 	validUntil   time.Time
 }
 
-func (s *memoryStore) AccessToken() string {
+// persistentStore implements tokenStore backed by disk.
+type persistentStore struct {
+	mu           sync.RWMutex // protects the following members
+	AccessToken  string
```

Comment:
Do these members need to be exported? It seems risky to export members which you expect to be protected by a mutex, especially since the mutex itself is unexported.